dronekit/__init__.py

```diff
@@ -1893,6 +1884,115 @@ def wait_ready(self, *types, **kwargs):
 
         return True
 
+class Gimbal(HasObservers):
+    """
+    Gimbal control and status.
+
+    An object of this type is returned by :py:attr:`Vehicle.gimbal`.
+    .. note::
+
+        All the orientation  "values" (e.g. ``gimbal.yaw``) are initially
+        created with value ``None``. The orientation values are populated
+        shortly after initialisation ONLY if a gimbal is present."""
+
+    def __init__(self, vehicle):
+        super(Gimbal, self).__init__()
+
+        self._pitch = None
+        self._roll = None
+        self._yaw = None
+        self._vehicle = vehicle
+
+        @vehicle.on_message('MOUNT_STATUS')
+        def listener(vehicle, name, m):
+            self._pitch = m.pointing_a / 100
+            self._roll = m.pointing_b / 100
+            self._yaw = m.pointing_c / 100
+            vehicle.notify_attribute_listeners('gimbal', vehicle.gimbal)
+
+    @property
+    def pitch(self):
+        return self._pitch
+
+    @property
+    def roll(self):
+        return self._roll
+
+    @property
+    def yaw(self):
+        return self._yaw
+
+    def rotate(self, pitch, roll, yaw):
+        """
+        Rotate the gimbal to a specific vector.
+
+        :param pitch: gimbal pitch, degrees
+        :param roll: gimbal roll, degrees
+        :param yaw: gimbal yaw, degrees in global frame
+        """
+        msg = self._vehicle.message_factory.mount_configure_encode(
+                    0, 1,    # target system, target component
+                    mavutil.mavlink.MAV_MOUNT_MODE_MAVLINK_TARGETING,  #mount_mode
+                    1,  # stabilize roll
+                    1,  # stabilize pitch
+                    1,  # stabilize yaw
+                    )
+        self._vehicle.send_mavlink(msg)
+        msg = self._vehicle.message_factory.mount_control_encode(
+                    0, 1,    # target system, target component
+                    pitch * 100, # pitch is in centidegrees
+                    roll * 100, # roll
+                    yaw * 100, # yaw is in centidegrees
+                    0) # save position
+        self._vehicle.send_mavlink(msg)
+
+    def target_gps(self,roi):
```

Comment:
This probably should support either type of global frame. For brownie points, would be great if it could also take a LocationLocal. @tcr3dr Pity we don't support the body frame for local, because it would be very handy to be able to say "set ROI 10metres in front of me"